src/ir/effects.h

```diff
@@ -78,15 +112,17 @@ struct EffectAnalyzer
   bool accessesMemory() const { return calls || readsMemory || writesMemory; }
 
   bool hasGlobalSideEffects() const {
-    return calls || globalsWritten.size() > 0 || writesMemory || isAtomic;
+    return calls || globalsWritten.size() > 0 || writesMemory || isAtomic ||
+           mayThrow;
```

Comment:
Does `hasGlobalSideEffects` mean that it is observable in other threads? If so yeah, I should move it to `hasSideEffects`.

Reply:
I think "global" means "non-local", that is, outside of locals in the function.

Yes, I think maybe just setting `branches = True` instead of where you set `mayThrow` may be enough.